x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/IndexResolver.java

```diff
@@ -84,58 +84,88 @@ public IndexResolver(Client client) {
+        String indexPattern,
         Set<String> fieldNames,
         QueryBuilder requestFilter,
         boolean includeAllDimensions,
         boolean useAggregateMetricDoubleWhenNotSupported,
         boolean useDenseVectorWhenNotSupported,
+        TransportVersion minimumVersion,
         ActionListener<IndexResolution> listener
     ) {
-        ActionListener<Versioned<IndexResolution>> ignoreVersion = listener.delegateFailureAndWrap(
+        ActionListener<Versioned<IndexResolution>> versionIgnoringListener = listener.delegateFailureAndWrap(
             (l, versionedResolution) -> l.onResponse(versionedResolution.inner())
         );
 
-        resolveAsMergedMappingAndRetrieveMinimumVersion(
-            indexWildcard,
+        resolveAsMergedMapping(
+            indexPattern,
             fieldNames,
             requestFilter,
             includeAllDimensions,
             useAggregateMetricDoubleWhenNotSupported,
             useDenseVectorWhenNotSupported,
-            ignoreVersion
+            minimumVersion,
+            versionIgnoringListener
         );
     }
 
     /**
-     * Resolves a pattern to one (potentially compound meaning that spawns multiple indices) mapping. Also retrieves the minimum transport
-     * version available in the cluster (and remotes).
+     * Perform a field caps request to resolve a pattern to one mapping (potentially compound, meaning it spans multiple indices).
+     * The field caps response contains the minimum transport version of all clusters that apply to the pattern,
+     * and it is used to deal with previously unsupported data types during resolution.
+     * <p>
+     * The retrieved minimum version is passed on to the listener.
+     * <p>
+     * If a field's type is not supported on the minimum version, it will be {@link DataType#UNSUPPORTED}.
      */
-    public void resolveAsMergedMappingAndRetrieveMinimumVersion(
-        String indexWildcard,
+    public void resolveAsMergedMapping(
+        String indexPattern,
+        Set<String> fieldNames,
+        QueryBuilder requestFilter,
+        boolean includeAllDimensions,
+        boolean useAggregateMetricDoubleWhenNotSupported,
+        boolean useDenseVectorWhenNotSupported,
+        ActionListener<Versioned<IndexResolution>> listener
+    ) {
+        resolveAsMergedMapping(
+            indexPattern,
+            fieldNames,
+            requestFilter,
+            includeAllDimensions,
+            useAggregateMetricDoubleWhenNotSupported,
+            useDenseVectorWhenNotSupported,
+            null,
+            listener
+        );
+    }
+
+    private void resolveAsMergedMapping(
+        String indexPattern,
         Set<String> fieldNames,
         QueryBuilder requestFilter,
         boolean includeAllDimensions,
         boolean useAggregateMetricDoubleWhenNotSupported,
         boolean useDenseVectorWhenNotSupported,
+        @Nullable TransportVersion minimumVersion,
```

Comment:
This `@Nullable` is a bit confusing. This method is private to the class and has only two usages:
* one in `resolveAsMergedMappingForVersion` that never supplies null
* one in `resolveAsMergedMapping` that allways supplies null

Lets change the second one to supply min compatible to avoid nullability and a condition below.

Reply:
Had to re-do it because I was wrong initially :D there's only 1 method now, and it has a nullable minimum version to be passed in, intentionally.